PROTOCOL.md

```diff
@@ -0,0 +1,347 @@
+`kubernetes`)
+3. the `targetsrv` that will supply the `keep` with its `items`
+
+We can achieve this using the same protocol as before, but by using a separate
+`targetsrv` process and some additional steps during the `Instantiation Phase`.
+
+### Phase #1: Instantiation
+
+Instantiating a `keep` under the `Remote Pull` model is basically the same as
+the `Remote Push` model, but with some additional steps at the beginning.
+
+```mermaid
+graph LR
+    subgraph tenant
+        A(operator)
+        B(targetsrv)
+    end
+
+    subgraph host
+        C(scheduler)
+        D(enarx run)
+        E(enarx srv)
+        F(keep)
+
+	    style F fill:#0f0,stroke:#333,stroke-width:4px
+    end
+
+    A -. "1. enarx push" .-> B
+    B -. "2. pubₜ" .-> A
+
+    A -. "3. configure [target = (uri, pubₜ)]" .-> A
+    A -. "4. schedule" .-> C
+
+    C -.-> D
+    D --> E
+    E -.-> F
+```
+
+1. The `operator` issues the `enarx push` command to store a fully deployable
+keep inside the `targetsrv`. The `targetsrv` receives the the following data:
+measurement of all allowed `shim`, `exec` and `backend` combinations, the
+`manifest` and all `items`.
+2. Upon receipt of the aforementioned items, the `targetsrv` creates a new keep
+instance record storing all the items and generates the target keypair (`pubₜ`,
+`prvₜ`), returning the public key (`pubₜ`) to the `operator`. The `enarx push`
+process returns an appropriate `target` configuration. This will likely have a
+`uri` of `tcp:...` so that it can connect to the `targetsrv`.
+3. The `operator` uses the `target` configuration to produce `targets` which it
+associates with the job to run. For example, the operator might include the
+`targets` in a `Dockerfile` and then build it with `docker build`.
+4. The `operator` schedules the job to run. For example, this might include
+configuring the aforementioned container to run under `kubernetes`.
+
+All remaining steps are identical to the `Remote Push` model.
+
+### Phase #2: Provisioning
+
+The provisioning phase is identical to the `Remote Push` model, with the
+exception that the `keep` will connect to the `targetsrv` instead of the `enarx
+run` instance. This is because the `target` contains a different `uri`.
+Further, the `targetsrv` will use the `pubₜ` key to uniquely identify which set
+of `items` to deploy. Other than these differences, this process is identical
+to the `Remote Push` model.
+
+## Model #3: Multi-Party Computation
+
+As an astute reader, you've probably noticed that we are passing `targets` into
+the `keep` instantiation process but that this has, until now, only had a
+single `target` configured. Under the multi-party computation model, we are
+going to assembly a composite `keep` containing `items` from multiple different
+parties. This model builds on the previous models by allowing the `keep` to
+contact multiple `targetsrv` instances and merging each of their `manifest`s
```

Comment:
Worth noting that multiple parties may share the same targetsrv, and place multiple targets within it (differentiated by URL and identified by public key)?